File: src/main/java/com/eager/questioncloud/creator/entity/CreatorEntity.java
package com.eager.questioncloud.creator.entity;

import com.eager.questioncloud.creator.domain.Creator;
import com.eager.questioncloud.question.vo.Subject;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "creator")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class CreatorEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Long userId;

    @Column
    @Enumerated(EnumType.STRING)
    private Subject mainSubject;

    @Column
    private String introduction;

    @Builder
    public CreatorEntity(Long id, Long userId, Subject mainSubject, String introduction) {
        this.id = id;
        this.userId = userId;
        this.mainSubject = mainSubject;
        this.introduction = introduction;
    }

    public Creator toModel() {
        return Creator.builder()
            .id(id)
            .userId(userId)
            .mainSubject(mainSubject)
            .introduction(introduction)
            .build();
    }
}

File: qc-domain/qc-creator/qc-creator-core/src/main/kotlin/com/eager/questioncloud/creator/infrastructure/entity/CreatorEntity.kt
package com.eager.questioncloud.creator.infrastructure.entity

import com.eager.questioncloud.creator.domain.Creator
import jakarta.persistence.*

@Entity
@Table(name = "creator")
class CreatorEntity(
    @GeneratedValue(strategy = GenerationType.IDENTITY) @Id var id: Long = 0,
    @Column var userId: Long,
    @Column var mainSubject: String,
    @Column var introduction: String,
) {
    fun toModel(): Creator {
        return Creator(id, userId, mainSubject, introduction)
    }

    companion object {
        fun from(creator: Creator): CreatorEntity {
            return CreatorEntity(creator.id, creator.userId, creator.mainSubject, creator.introduction)
        }
    }
}
